tests/Ylmish.Tests/Y.Delta.fs

```diff
@@ -287,39 +273,69 @@ let tests = testList "Y.Delta" [
             let ytext = ydoc.getText "test"
             let _ = ytext.insert(0, "0123456789")
-            ytext.applyDelta(delta)
+
+            Y.Text.Impl.applyAdaptiveDelta list delta ytext
 
             Expect.equal (ytext.toString()) "abc01456789" "ytext doesn't equal expected value"
+        }
+        test "applyAdaptiveDelta del 2, insert 'efg'" {
+            let input = [
+                0, ElementOperation.Remove
+                1, ElementOperation.Remove
+                2, ElementOperation.Set 'a'
+                3, ElementOperation.Set 'b'
+                4, ElementOperation.Set 'c'
+            ]
+            let list, delta = toIndexListDelta input
 
-            Expect.equal (List.ofSeq delta) [
-                Y.Delta.Insert ("abc")
-                Y.Delta.Retain 2
-                Y.Delta.Delete 2
-            ] ""
+            let ydoc = Y.Doc.Create ()
+            let ytext = ydoc.getText "test"
+            let _ = ytext.insert(0, "0123456789")
+
+            Y.Text.Impl.applyAdaptiveDelta list delta ytext
+
+            Expect.equal (ytext.toString()) "abc23456789" "ytext doesn't equal expected value"
         }
-        test "ofAdaptive given \"abd\", insert 'c' between the 'b' and the 'd'" {
+        test "applyAdaptiveDelta ins 'abc', ret 2, del 2, insert 'efg'" {
+            let input = [
+                0, ElementOperation.Set 'a'
+                1, ElementOperation.Set 'b'
+                2, ElementOperation.Set 'c'
+                // 3
+                // 4
+                5, ElementOperation.Remove
+                6, ElementOperation.Remove
+                7, ElementOperation.Set 'e'
+                8, ElementOperation.Set 'f'
+                9, ElementOperation.Set 'g'
+            ]
+            let list, delta = toIndexListDelta input
+
+            let ydoc = Y.Doc.Create ()
+            let ytext = ydoc.getText "test"
+            let _ = ytext.insert(0, "0123456789")
+
+            Y.Text.Impl.applyAdaptiveDelta list delta ytext
+
+            Expect.equal (ytext.toString()) "abc01efg456789" "ytext doesn't equal expected value"
+        }
+        test "applyAdaptiveDelta given \"abd\", insert 'c' between the 'b' and the 'd'" {
             let input = [
                 2, ElementOperation.Set 'c'
             ]
             let list, delta = toIndexListDelta input
-            let delta = Y.Text.Impl.ofAdaptive list delta
 
             let ydoc = Y.Doc.Create ()
             let ytext = ydoc.getText "test"
             let _ = ytext.insert(0, "abd")
-            ytext.applyDelta(delta)
 
-            Expect.equal (ytext.toString()) "abcd" "ytext doesn't equal expected value"
+            Y.Text.Impl.applyAdaptiveDelta list delta ytext
 
-            Expect.equal (List.ofSeq delta) [
-                Y.Delta.Retain   2
-                Y.Delta.Insert ("c")
-            ] ""
+            Expect.equal (ytext.toString()) "abcd" "ytext doesn't equal expected value"
         }
     ]
  ]

```

Comment:
I think this PR could also remove `src/Ylmish/Adaptive.Index.fs`, `tests/Ylmish.Tests/Adaptive.Index.fs`, and tests "Index.at and clist initialisation return the same indices" and "Index.at and clist.InsertAt return the same indices" of `tests/Ylmish.Tests/Y.Delta.fs`.

Reply:
(This comment isn't related to these lines... I don't know how to make comments on GitHub that are _part_ of a review but not against a specific line)

Reply:
> I think this PR could also remove ...

Agreed (and I've done so).